### PR TITLE
Update generated docs script

### PR DESCRIPTION
**Pull request by clarketm**
Please provide a description for what this PR is for.

Update reference docs script: `grab_reference_docs.sh` based on: https://github.com/istio/istio/pull/21837.

ref: https://prow.istio.io/view/gcs/istio-prow/logs/update-ref-docs_istio.io_periodic/53

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: scripts/grab_reference_docs.sh [hunk 1]
```diff
@@ -48,7 +48,6 @@ COMPONENTS=(
     https://github.com/istio/istio.git@"${SOURCE_BRANCH_NAME}"@istioctl/cmd/istioctl@istioctl
     https://github.com/istio/istio.git@"${SOURCE_BRANCH_NAME}"@pilot/cmd/pilot-agent@pilot-agent
     https://github.com/istio/istio.git@"${SOURCE_BRANCH_NAME}"@pilot/cmd/pilot-discovery@pilot-discovery
-    https://github.com/istio/istio.git@"${SOURCE_BRANCH_NAME}"@security/cmd/istio_ca@istio_ca
```
Comment (frankbu) [inline scripts/grab_reference_docs.sh]: Is there doc moved, or need to be added, somewhere else that documents how to set some of the CA options? For example Istiod cmd options for CA?

Comment (clarketm) [reply scripts/grab_reference_docs.sh]: Not certain, I will defer to @howardjohn since the deletion was his change.

Comment (howardjohn) [reply scripts/grab_reference_docs.sh]: The whole binary is gone. Any settings that can be used to configure Istiod CA should automagically show up in the pilot-discovery page

**Pull request merged**